File: result.cpp
#include "ros/ros.h"
#include "beginner_tutorials/data.h"
#include "beginner_tutorials/info.h"

void chatterCallback(const beginner_tutorials::info::ConstPtr& msg)
{
  ROS_INFO("BMI_ID : [%d]", msg->ID);
  ROS_INFO("BMI : [%f]", msg->BMI);
  //pub->publish(ID); 
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "result");
	ros::NodeHandle n;

	ros::Subscriber sub = n.subscribe("BMI_ID", 1000, chatterCallback); /*chatter->*/
	ros::spin();

	return 0;
}

File: result.py
#!/usr/bin/env python
import rospy
from beginner_tutorials.msg import data
from beginner_tutorials.msg import info

def callback(data):
	rospy.loginfo("ID: %d" % (data.ID))
	rospy.loginfo("BMI: %f" % (data.BMI))
	
def listener():
    rospy.init_node('result_listener', anonymous=True)
    rospy.Subscriber("result", info, callback)

    # spin() simply keeps python from exiting until this node is stopped
    rospy.spin()

if __name__ == '__main__':
    listener()